data/libs/EquipType.lua

```diff
@@ -32,166 +31,278 @@ local misc = {}
+	local old = type.New
+	local un = type.Unserialize
+
+	-- Create a new metatable for instances of the prototype object;
+	-- delegates serialization to the base class of the proto
+	function type.New(...)
+		local inst = old(...)
+		inst.meta = utils.mixin(type.meta, { __index = inst })
+		return inst
+	end
+
+	function type.Unserialize(inst)
+		inst = un(inst) ---@type any
+
+		-- if we have a "__proto" field we're an instance of the equipment prototype
+		if rawget(inst, "__proto") then
+			setmetatable(inst, inst.__proto.meta)
+		end
+
+		return inst
 	end
-	return num
 end
 
-local function __ApplyCapabilities(ship, capabilities, num, factor)
-	if num <= 0 then return 0 end
-	factor = factor or 1
-	for k,v in pairs(capabilities) do
-		local full_name = k.."_cap"
-		local prev = (ship:hasprop(full_name) and ship[full_name]) or 0
-		ship:setprop(full_name, (factor*v*num)+prev)
+function EquipType:Serialize()
+	local tmp = EquipType.Super().Serialize(self)
+	local ret = {}
+	for k,v in pairs(tmp) do
+		if type(v) ~= "function" then
+			ret[k] = v
+		end
 	end
-	return num
+
+	ret.transient = nil
+	return ret
 end
 
-function EquipType:Install(ship, num, slot)
-	local caps = self.capabilities
-	num = __ApplyMassLimit(ship, caps, num)
-	return __ApplyCapabilities(ship, caps, num, 1)
+function EquipType.Unserialize(data)
+	local obj = EquipType.Super().Unserialize(data)
+	setmetatable(obj, EquipType.meta)
+
+	-- Only patch the common prototype with runtime transient data
+	if EquipType.isProto(obj) then
+		EquipType._createTransient(obj)
+	end
+
+	return obj
 end
 
-function EquipType:Uninstall(ship, num, slot)
-	return __ApplyCapabilities(ship, self.capabilities, num, -1)
+-- Method: GetName
+--
+-- Returns the translated name of this equipment item suitable for display to
+-- the user.
+---@return string
+function EquipType:GetName()
+	return self.transient.name
 end
 
+-- Method: GetDescription
+--
+-- Returns the translated description of this equipment item suitable for
+-- display to the user
+---@return string
+function EquipType:GetDescription()
+	return self.transient.description
+end
+
+--==============================================================================
+
 -- Base type for weapons
-local LaserType = utils.inherits(EquipType, "LaserType")
-function LaserType:Install(ship, num, slot)
-	if num > 1 then num = 1 end -- FIXME: support installing multiple lasers (e.g., in the "cargo" slot?)
-	if LaserType.Super().Install(self, ship, 1, slot) < 1 then return 0 end
-	local prefix = slot..'_'
-	for k,v in pairs(self.laser_stats) do
-		ship:setprop(prefix..k, v)
+---@class Equipment.LaserType : EquipType
+---@field laser_stats table
+local LaserType = utils.inherits(EquipType, "Equipment.LaserType")
+
+---@param ship Ship
+---@param slot HullConfig.Slot
+function LaserType:OnInstall(ship, slot)
+	EquipType.OnInstall(self, ship, slot)
+
+	for k, v in pairs(self.laser_stats) do
+		-- TODO: allow installing more than one laser
```

Comment:
Isn't this already possible with the various front weapon slots?

HARDPOINT_WEAPON_FRONT, HARDPOINT_WEAPON_CHIN, HARDPOINT_WEAPON_FRONT_(LEFT|RIGHT), HARDPOINT_WEAPON_FRONT_(LEFT|RIGHT)_NOSE, etc?

Reply:
No. The weapon slots are named that way in anticipation of future work during Stage 2, but the underlying implementation (`FixedGuns.cpp`) only handles a single front/rear weapon slot. I have an in-progress branch (since about 3 years ago) that fully replaces that implementation with one that can handle multiple weapons and can be extended to support turrets, ammunition-consuming weapons, etc.